zokrates_core/src/semantics.rs

```diff
@@ -1121,13 +1138,24 @@ impl<'ast, T: Field> Checker<'ast, T> {
                 }
             }
             Expression::Identifier(name) => {
-                // check that this generic parameter is defined
-                match generics_map.get(&name) {
-                    Some(index) => Ok(Constant::Generic(GenericIdentifier {name, index: *index})),
-                    None => Err(ErrorInner {
-                                    pos: Some(pos),
-                                    message: format!("Undeclared generic parameter in function definition: `{}` isn\'t declared as a generic constant", name)
-                                })
+                match (constants_map.get(name), generics_map.get(&name)) {
+                    (Some(c), None) => {
+                        match c {
+                            Type::Uint(bitwidth) => Ok(Constant::Identifier(name, bitwidth.to_usize())),
```

Comment:
This should only accept constants of bitwidth 32 for now. There should be a test for this similar to the one with field.
```suggestion
                            Type::Uint(UBitwidth::B32) => Ok(Constant::Identifier(name, bitwidth.to_usize())),
```

Reply:
I was doing a bit of future-proofing here, but I agree this arm should check only for 32 now